### ISSUE TITLE
Set a User Agent when making requests to Bugzilla's API

### ISSUE THREAD

**Issue by edmorley** [mozilla/mediawiki-bugzilla#84]
It's best practice to set an informative User Agent when making requests to an external API, and helps those maintaining the API track the sources of inefficient queries/heavy load :-)

**Comment by rdalverny** [comment]
This would do but to be sure before making a PR...

* I would define a version for the extension. Is it ok? I am not sure this (global `$wgBugzillaExtVersion`) is the best place to put it, though.
* reporting both MediaWiki and PHP versions may help too.

```diff
diff --git a/Bugzilla.php b/Bugzilla.php
index 2c48a59..8203874 100755
--- a/Bugzilla.php
+++ b/Bugzilla.php
@@ -173,6 +173,8 @@ function BugzillaRender($input, array $args, Parser $parser, $frame=null ) {
  * compatibility.
  */
 
+$wgBugzillaExtVersion = '1.0.0';
+
 // Remote API
 $wgBugzillaRESTURL     = 'https://bugzilla.mozilla.org/bzapi'; // The URL for your Bugzilla API installation
 $wgBugzillaURL         = 'https://bugzilla.mozilla.org'; // The URL for your Bugzilla installation
diff --git a/BugzillaQuery.class.php b/BugzillaQuery.class.php
index 07f7fe8..b02d40d 100644
--- a/BugzillaQuery.class.php
+++ b/BugzillaQuery.class.php
@@ -185,6 +185,9 @@ class BugzillaRESTQuery extends BugzillaBaseQuery {
     // Load data from the Bugzilla REST API
     public function _fetch_by_options() {
 
+        global $wgVersion;
+        global $wgBugzillaExtVersion;
+
         // Set up our HTTP request
         $options_array = array();
 
@@ -203,6 +206,8 @@ class BugzillaRESTQuery extends BugzillaBaseQuery {
         $request->setHeader('Accept', 'application/json');
         $request->setHeader('Content-Type', 'application/json');
 
+        $request->setHeader('User-Agent', 'MediawikiBugzilla/'.$wgBugzillaExtVersion.' MediaWiki/'.$wgVersion.' PHP/'.PHP_VERSION);
+
         // Save the real options
         $saved_options = $this->options;
```

**Comment by dennisroczek** [comment]
Better add the 'Äversion' to the $wgExtensionCredits array in line https://github.com/mozilla/mediawiki-bugzilla/blob/master/Bugzilla.php#L21 and load that, see https://www.mediawiki.org/wiki/Manual:Developing_extensions#Registering_features_with_MediaWiki

So that version will also be displayed at the [[Special:Version]] page on mediawiki itself.

**Comment by rdalverny** [comment]
Looks like a good place to add it, but how do I access this value later? The credits array is appended to `$wgExtensionCredits` without a specific index...

One could define a class constant in `Bugzilla.class.php` but that would mean loading it before. Or define a constant just before (`BUGZILLA_EXT_VERSION`), add it to `$wgExtensionCredits` and use it in the `setHeader` function. But that doesn't feel right.

Looking into other extensions to see how they manage this (although most recent I find use the new `extension.json` file for that).

**Comment by dennisroczek** [comment]
Yeah, we still have to update the extension to use the new registration system for extensions which was introduced in MW1.25. :-(

yeah, you're right. I remember that I had a similar situation seen with different extensions. Define a global and use that as you have described it.

**Comment by LegNeato** [comment]
👍 Fixed in #85, thanks @rdalverny!
